dask/array/tests/test_fft.py

```diff
@@ -0,0 +1,230 @@
+import unittest
+
+import numpy as np
+import numpy.fft as npfft
+
+import dask
+from dask.array.core import Array
+from dask.utils import raises
+import dask.array as da
+from dask.array.fft import fft, ifft, rfft, irfft, hfft, ihfft
+
+
+def mismatch_err(mismatch_type, got, expected):
+    return '%s mismatch, got %s, expected %s' % (mismatch_type, got, expected)
+
+
+def eq(a, b):
+    assert a.shape == b.shape, mismatch_err('shape', a.shape, b.shape)
+
+    if isinstance(a, Array):
+        adt = a._dtype
+        a = a.compute(get=dask.get)
+    else:
+        adt = getattr(a, 'dtype', None)
+    if isinstance(b, Array):
+        bdt = b._dtype
+        b = b.compute(get=dask.get)
+    else:
+        bdt = getattr(b, 'dtype', None)
+
+    assert str(adt) == str(bdt), mismatch_err('dtype', str(adt), str(bdt))
+
+    try:
+        return np.allclose(a, b)
+    except TypeError:
+        pass
+
+    c = a == b
+
+    if isinstance(c, np.ndarray):
+        return c.all()
+    else:
+        return c
+
+
+class TestFFT(unittest.TestCase):
+    nparr = np.arange(100).reshape(10, 10)
+    darr = da.from_array(nparr, chunks=(1, 10))
+    darr2 = da.from_array(nparr, chunks=(10, 1))
+
+    def test_cant_fft_chunked_axis(self):
+        bad_darr = da.from_array(self.nparr, chunks=(5, 5))
+
+        assert raises(ValueError, lambda: fft(bad_darr))
+        assert raises(ValueError, lambda: fft(bad_darr, axis=0))
+
+    def test_fft(self):
+        res = fft(self.darr)
+        expected = npfft.fft(self.nparr)
+        assert eq(res, expected)
+
+        res2 = fft(self.darr2, axis=0)
+        expected2 = npfft.fft(self.nparr, axis=0)
+        assert eq(res2, expected2)
```

Comment:
For brevity's sake I wonder if these can be made more compact.  This might make it a bit easier to visually check

```python
assert eq(fft(self.darr), npfft.fft(self.nparr))
assert eq(fft(self.darr2, axis=0), npfft.fft(self.nparr, axis=0))
```